tools/testsys/src/aws_k8s.rs

```diff
@@ -0,0 +1,155 @@
+use crate::aws_resources::{ami, ec2_crd, get_ami_id, migration_crd};
+use crate::crds::{
+    BottlerocketInput, CrdCreator, CrdInput, CreateCrdOutput, MigrationInput, OrchestratorInput,
+    TestInput,
+};
+use crate::error::{self, Result};
+use crate::sonobuoy::sonobuoy_crd;
+use bottlerocket_types::agent_config::{
+    ClusterType, CreationPolicy, EksClusterConfig, EksctlConfig, K8sVersion,
+};
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+use maplit::btreemap;
+use model::constants::NAMESPACE;
+use model::{Agent, Configuration, Crd, DestructionPolicy, Resource, ResourceSpec};
+use snafu::{OptionExt, ResultExt};
+use std::str::FromStr;
+
+/// A `CrdCreator` responsible for creating crd related to `aws-k8s` variants.
+pub(crate) struct AwsK8sCreator {
+    pub(crate) region: String,
+    pub(crate) ami_input: String,
+    pub(crate) migrate_starting_commit: Option<String>,
+}
+
+#[async_trait::async_trait]
+impl CrdCreator for AwsK8sCreator {
+    /// Determine the AMI from `amis.json`.
+    fn image_id(&self, _: &CrdInput) -> Result<String> {
+        ami(&self.ami_input, &self.region)
+    }
+
+    /// Determine the starting image from EC2 using standard Bottlerocket naming conventions.
+    async fn starting_image_id(&self, crd_input: &CrdInput) -> Result<String> {
+        get_ami_id(
+            format!(
+                "bottlerocket-{}-{}-{}-{}",
+                crd_input.variant, crd_input.arch, crd_input.starting_version.as_ref().context(error::InvalidSnafu{what: "The starting version must be provided for migration testing"})?, self.migrate_starting_commit.as_ref().context(error::InvalidSnafu{what: "The commit for the starting version must be provided if the starting image id is not"})?
+            ), &crd_input.arch,
+           & self.region,
+        )
+        .await
+    }
+
+    /// Create an EKS cluster CRD with the `cluster_name` in `orchestrator_input`.
+    async fn orchestrator_crd<'a>(
+        &self,
+        orchestrator_input: OrchestratorInput<'a>,
+    ) -> Result<CreateCrdOutput> {
+        let labels = orchestrator_input.crd_input.labels(btreemap! {
+            "testsys/type".to_string() => "cluster".to_string(),
+            "testsys/cluster".to_string() => orchestrator_input.cluster_name.to_string(),
+            "testsys/region".to_string() => self.region.clone()
+        });
+
+        // Check if the cluster already has a crd
+        if let Some(cluster_crd) = orchestrator_input
+            .crd_input
+            .existing_crds(
+                &labels,
+                &["testsys/cluster", "testsys/type", "testsys/region"],
```

Comment:
Maybe we should think about constants for these. Technically even the array could be a constant but maybe that's not necessary.

Reply:
I think in this case, it's not necessary.